dm/pkg/binlog/event/util.go

```diff
@@ -354,6 +354,22 @@ func GetParserForStatusVars(statusVars []byte) (*parser.Parser, error) {
 	return parser2, err
 }
 
+// GetServerCollationByStatusVars gets server collation by binlog statusVars.
+func GetServerCollationByStatusVars(statusVars []byte) (string, error) {
+	vars, err := statusVarsToKV(statusVars)
+	b, ok := vars[QCharsetCode]
```

Comment:
maybe check err!=nil first?

Reply:
it reference `getSQLMode`, maybe its original intention is try to get `QCharsetCode` even has err, because the err may occur in parse other field in `statusVars`.
https://github.com/pingcap/ticdc/blob/master/dm/pkg/binlog/event/util.go#L331